basic_network.go

```diff
@@ -9,40 +9,51 @@ package basicnet
 import (
 	"context"
 	"errors"
+	"strings"
 
 	peerstore "gx/ipfs/QmPgDWmTmuzvP7QE5zwo1TmjbJme9pmZHNujB2453jkCTr/go-libp2p-peerstore"
 	ma "gx/ipfs/QmXY77cVe7rVRQXZZQRioukUM7aRW3BTcAgJe12MCtb3Ji/go-multiaddr"
 	peer "gx/ipfs/QmXYjuNuxVzXKJCfWasQk1RqkhVLDM9jtUKhqc2WPQmFSB/go-libp2p-peer"
 	protocol "gx/ipfs/QmZNkThpqfVXs9GNbexPrfBbXSLNYeKrE7jwFM2oqHbyqN/go-libp2p-protocol"
 	net "gx/ipfs/QmahYsGWry85Y7WUe2SX5G4JkH2zifEQAUtJVLZ24aC9DF/go-libp2p-net"
 
+	"github.com/ericxtang/m3u8"
 	"github.com/golang/glog"
 	lpnet "github.com/livepeer/go-livepeer/net"
-	"github.com/livepeer/go-livepeer/types"
 )
 
 var Protocol = protocol.ID("/livepeer_video/0.0.1")
 var ErrNoClosePeers = errors.New("NoClosePeers")
 var ErrUnknownMsg = errors.New("UnknownMsgType")
 var ErrProtocol = errors.New("ProtocolError")
 var ErrTranscodeResult = errors.New("TranscodeResultError")
+var ErrGetMasterPlaylist = errors.New("ErrGetMasterPlaylist")
 
 type VideoMuxer interface {
 	WriteSegment(seqNo uint64, strmID string, data []byte) error
 }
 
 //BasicVideoNetwork implements the VideoNetwork interface.  It creates a kademlia network using libp2p.  It does push-based video delivery, and handles the protocol in the background.
 type BasicVideoNetwork struct {
-	NetworkNode  *NetworkNode
-	broadcasters map[string]*BasicBroadcaster
-	subscribers  map[string]*BasicSubscriber
-	relayers     map[string]*BasicRelayer
+	NetworkNode          *NetworkNode
+	broadcasters         map[string]*BasicBroadcaster
+	subscribers          map[string]*BasicSubscriber
+	relayers             map[string]*BasicRelayer
+	mplMap               map[string]*m3u8.MasterPlaylist
+	mplChans             map[string]chan *m3u8.MasterPlaylist
+	transResultCallbacks map[string]func(transcodeResult map[string]string)
```

Comment:
In general it was unclear to me what a `TranscodeResult` is supposed to represent. It sounds like it's supposed to be the metadata about a new transcoded rendition of the stream that's returned to the broadcaster. Maybe we should rename it to something like `TranscodeResponse` or `TranscodedStreamMetadata`, or we could just comment what it's supposed to represent.